src/mission/odperformance.jl

```diff
@@ -1,352 +1,478 @@
+        KAfTE   = ac.parad[iaKAfTE, ipcruise1]
+
+    # Zero-fuel weight for this mission
+        Wzero   = ac.parg[igWMTO] - ac.parg[igWfuel] - ac.parg[igWpay] + ac.parmd[imWpay]  #This ensures that this will work for multi-mission in the future
+        Wempty  = ac.parg[igWMTO] - ac.parg[igWfuel] - ac.parg[igWpay]
+    # Payload fraction for this mission
+        rpay = ac.parmd[imWpay]/ac.parg[igWpay]
+        ξpay = 0.
+
+    S    = ac.wing.layout.S
+    # Convert FLs to altitude in m
+    alts = FL*100*ft_to_m
+
+    # Initialize arrays
+    N = length(FL)
+    T0s = zeros(Float64, N)
+    p0s = zeros(Float64, N)
+    ρ0s = zeros(Float64, N)
+    a0s = zeros(Float64, N)
+    μ0s = zeros(Float64, N)
+    V0s = zeros(Float64, N)
+    Reunits = zeros(Float64, N)
+    M0s = zeros(Float64, N)
+    Wbouys = zeros(Float64, N)
+    Ws = zeros(Float64, N)
+    γs = ones(Float64, N)*0.015
+    Tt41s = zeros(Float64, N)
+
+    deNOx = zeros(Float64, N)
+    mdotf = zeros(Float64, N)
+    ROC   = zeros(Float64, N)
+    ROCmaxcrz = zeros(Float64, N)
+    FFmaxcrz  = zeros(Float64, N)
+    EGTcrz = zeros(Float64, N)
+
+    iceil = N
+    crzmdotf = zeros(Float64, N)
+    crzEINOx = zeros(Float64, N)
+    crzFAR   = zeros(Float64, N)
+    clmbEINOx = zeros(Float64, N)
+    crzTAS   = zeros(Float64, N)
+    desTAS   = zeros(Float64, N)
+    Tt41crz   = zeros(Float64, N)
+    Tt41crzmax   = zeros(Float64, N)
+
+    @inbounds for i =1:N
+        T0s[i], p0s[i], ρ0s[i], a0s[i], μ0s[i] = atmos(alts[i]/1000)
+        rhocab = max( ac.parg[igpcabin] , p0s[i] ) / (RSL*TSL) #Should be T0s to be more accurate?
+        Wbouys[i] = (rhocab-ρ0s[i])*gee*ac.parg[igcabVol]
     end
 
-    ρ  = ρ0s[i]
-    μ  = μ0s[i]
-    Vsound = a0s[i]
-    gamV = 0.0
-    cosg = cos(gamV)
-    γs[i] = gamV
-    BW   = W + Wbouys[i]
-
-    dgamV  = 1.0
-    Ftotal = 0.0
-    DoL    = 0.0
-    V      = 0.0
-
-    if (Ldebug)
-        printstyled(@sprintf("\t%5s  %10s  %10s  %10s  %10s  %10s  %10s  %10s  %10s  %10s  %10s  %10s  %10s\n",
-                        "iterg", "dgamV", "gamV", "cosg", "CL", "BW", "Tt4", "Ftotal", "DoL", "V", "Alt", "M", "Tt4"); color = :light_green)
-    end
-    for  iterg = 1:itergmax
-
-        q = 0.5*ρ*TAScl^2
-        CL = BW*cosg/(q*S)
-        para[iaCL, ip] = CL
-
-        V = TAScl
-        Mach = V/Vsound
-        para[iaMach, ip] = Mach
-
-        V0s[i] = V # Save climb TAS (m/s) to V0s
-        desTAS[i] = TASdes/kts_to_mps # Save descent TAS separately, convert to knots
-        M0s[i] = Mach
-        Reunits[i] = V*ρ/μ
+    MNcr = ac.parad[iaMach, ipcruise1]
+    Tt41max = maximum(ac.pared[ieTt41, :])  #TODO set to max temp
+    Tt41CR = ac.pared[ieTt41,ipcruise1]
+    println(@sprintf("%12s %12s %12s %12s %12s %12s %12s %12s %12s %12s %12s %12s %12s %12s", 
+    "FL", "TAS", "CAS", "Mach", "Fn", "L/D", "Tt4max", "Tmetmax", "FFmax", "Tt4cruise", "Tmetcruise", "FFcruise", "CL", "CLh"))
+    # integrate trajectory over climb
+    for   i = 1:N
```

Comment:
labelled indices are nice
```suggestion
    for   iFL = 1:N
```